tests/scalar_gadgets_tests.rs

```diff
@@ -4,238 +4,233 @@
+    // This should pass since we sent 0 as selector and the circuit closure is constraining the
+    // result to be equal to 0.
+    assert!(verifier.verify(&proof, &vk, &[BlsScalar::zero()]).is_ok());
+
+    // Selector set to 1 shouldn't assign 0.
+    // Proving
+    prover.clear_witness();
+    circuit(
+        prover.mut_cs(),
+        BlsScalar::random(&mut rand::thread_rng()),
+        BlsScalar::one(),
+    );
+    let proof = prover.prove(&ck)?;
+    // This shouldn't pass since we sent 1 as selector and the circuit closure is constraining the
+    // result to be equal to 0 while the value assigned is indeed the randomly generated one.
+    assert!(verifier.verify(&proof, &vk, &[BlsScalar::zero()]).is_err());
+
+    Ok(())
+}
+
+#[test]
+fn test_conditionally_select_1() -> Result<(), Error> {
+    // The circuit closure runs the conditionally_select_one fn and constraints the result
+    // to actually to be equal to the provided expected_result.
+    let circuit = |composer: &mut StandardComposer,
+                   value: BlsScalar,
+                   selector: BlsScalar,
+                   expected_result: BlsScalar| {
+        let value = composer.add_input(value);
+        let selector = composer.add_input(selector);
+        let res = conditionally_select_one(composer, value, selector);
+        composer.constrain_to_constant(res, BlsScalar::zero(), Some(-expected_result));
+    };
+
+    // Generate Composer & Public Parameters
+    let pub_params = PublicParameters::setup(1 << 8, &mut rand::thread_rng())?;
+    let (ck, vk) = pub_params.trim(1 << 7)?;
+
+    // Selector set to 0 should asign 1
+    // Proving
+    let mut prover = Prover::new(b"testing");
+    circuit(
+        prover.mut_cs(),
+        BlsScalar::random(&mut rand::thread_rng()),
+        BlsScalar::zero(),
+        BlsScalar::one(),
+    );
+    let mut pi = prover.mut_cs().construct_dense_pi_vec().clone();
+    prover.preprocess(&ck)?;
+    let proof = prover.prove(&ck)?;
+
+    // Verification
+    let mut verifier = Verifier::new(b"testing");
+    circuit(
+        verifier.mut_cs(),
+        BlsScalar::random(&mut rand::thread_rng()),
+        BlsScalar::zero(),
+        BlsScalar::one(),
+    );
+    verifier.preprocess(&ck)?;
+    // This should pass since we sent 0 as selector and the circuit should then return
+    // 1.
+    assert!(verifier.verify(&proof, &vk, &pi).is_ok());
+
+    // Selector set to 1 should assign the randomly-generated value.
+    // Proving
+    prover.clear_witness();
+    let rand = BlsScalar::random(&mut rand::thread_rng());
+    circuit(prover.mut_cs(), rand, BlsScalar::one(), rand);
+    pi = prover.mut_cs().construct_dense_pi_vec().clone();
+    let proof = prover.prove(&ck)?;
+    // This should pass since we sent 1 as selector and the circuit closure should assign the randomly-generated
+    // value as a result.
+    verifier.verify(&proof, &vk, &pi)
+}
+
+#[test]
+fn test_is_not_zero() -> Result<(), Error> {
```

Comment:
Not 100% clear why we still use the old scheme of circuits here

Reply:
Legacy code we should address at some point I agree.. Let's file an issue for it tomorrow!